.github/workflows/ci.yml

```diff
@@ -110,7 +119,7 @@ jobs:
     - name: Build ${{ env.PACKAGE_NAME }} + consumers
       run: |
         python -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
-        python builder.pyz build -p ${{ env.PACKAGE_NAME }} downstream
+        python builder.pyz build -p ${{ env.PACKAGE_NAME }} --target windows-${{ matrix.arch }} --compiler msvc-14 downstream
```

Comment:
tldr; this change doesn't fix it. Don't let it hang up this PR, but we do need to fix it eventually...

These windows build jobs still aren't doing what they claim. Check the build output of:
- [windows-vc16 (x64)](https://github.com/awslabs/aws-crt-dotnet/actions/runs/3697094740/jobs/6261662833#step:3:185)
- [windows-vc14 (x64)](https://github.com/awslabs/aws-crt-dotnet/actions/runs/3697094740/jobs/6261662659#step:3:185)
- [windows-vc14 (x86)](https://github.com/awslabs/aws-crt-dotnet/actions/runs/3697094740/jobs/6261662489#step:3:179)

They're ALL doing the same exact thing, which is an x64 build using Visual Studio 2019
>   -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x86/cl.exe - skipped

that `ilammy/msvc-dev-cmd` action doesn't seem to actually do anything for us.

Also, passing `--compiler msvc-14` to aws-crt-builder apparently doesn't do what it claims.

Here's how get [aws-crt-cpp](https://github.com/awslabs/aws-crt-cpp/blob/b26ae59302d614f95cb41c6df62e83c136f18c5b/.github/workflows/ci.yml#L148-L159) to actually build with the MSVC 2015 (aka vs14, aka the 14.0 toolset, aka the versioning of Visual Studio is [super confusing](https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B#Internal_version_numbering))